cmd/snap/cmd_snapshot.go

```diff
@@ -0,0 +1,303 @@
+package main
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/jessevdk/go-flags"
+
+	"github.com/snapcore/snapd/client"
+	"github.com/snapcore/snapd/i18n"
+	"github.com/snapcore/snapd/strutil"
+	"github.com/snapcore/snapd/strutil/quantity"
+)
+
+type timeOpts struct {
+	AbsTime bool `long:"absolute-times"`
+	RelTime bool `long:"relative-times"`
+}
+
+var timeDescs = mixinDescs{
+	"absolute-times": i18n.G("Always display absolute times (in YYYY-MM-DD HH:MM format)."),
+	"relative-times": i18n.G("Always display relative times. If neither absolute nor relative times are requested, relative times are used for up to 30 days, and then absolute times in YYYY-MM-DD format."),
```

Comment:
Nice feature!

I suggest cutting down the verbosity of the help a bit. Perhaps:

- "Display times in YYYY-MM-DD HH:MM format."
- "Display relative times up to 30 days, then YYYY-MM-DD."

That said, can we cook formats so that the command output can be more easily parsed?  In other commands we have tried (or I have at least) to keep the columns without internal spacing, or at least the columns before the last one. It looks like it would be easy to do that here as well.. relative dates can be displayed similar to how Go formats time.Duration (e.g 20sec, 10min, 2h, 3days, etc), and for the absolute we can just use the typical RFC3339.

Reply:
After a brief discussion on IRC, agreed about dropping the whitespace, will do.
The advantage of `quantity.FormatDuration` is that the output is fixed width, and internationalised, so I'll stick with it if you don't mind.